apps/src/music/player/MusicLibrary.ts

```diff
@@ -26,15 +31,49 @@ async function loadLibrary(libraryName: string): Promise<MusicLibrary> {
       localLibrary as LibraryJson
     );
   } else {
-    const libraryJsonResponse = await HttpClient.fetchJson<LibraryJson>(
+    const libraryJsonResponsePromise = HttpClient.fetchJson<LibraryJson>(
       getBaseAssetUrl() +
         libraryFilename +
         '.json' +
         (requestVersion ? `?version=${requestVersion}` : ''),
       {},
       LibraryValidator
     );
-    return new MusicLibrary(libraryName, libraryJsonResponse.value);
+    const promises: Promise<GetResponse<Translations | LibraryJson>>[] = [
+      libraryJsonResponsePromise,
+    ];
+
+    const jsLocale = currentLocale().toLowerCase().replace('-', '_');
+    if (jsLocale !== 'en_us') {
+      const translationPromise = HttpClient.fetchJson<Translations>(
+        getBaseAssetUrl() + libraryFilename + '-loc/' + jsLocale + '.json'
+      );
+      promises.push(translationPromise);
+    }
+
+    // translations will be undefined if locale is en_us.
+    const [libraryJsonResponse, translations] = await Promise.allSettled(
+      promises
+    );
+
+    let libraryJson = {} as LibraryJson;
+    if (libraryJsonResponse.status === 'fulfilled') {
+      libraryJson = libraryJsonResponse.value.value as LibraryJson;
+    }
+
+    // Early return with no translations unless experiment is enabled for now.
+    if (!experiments.isEnabled('libraryLocalization')) {
```

Comment:
Can we use `isEnabledAllowingQueryString` so that it's also possible to see this without turning on the experiment in session storage?

Reply:
Oh slick I've never used this, will update!

Reply:
best thing I ever added :D